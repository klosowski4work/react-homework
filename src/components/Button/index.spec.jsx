import React from 'react';
import renderer from 'react-test-renderer';
import { mount, simulate } from 'enzyme';
import { Button } from './';


let props;
let wrapper;
describe('Button component', () => {
    beforeEach(() => {
        props = {
            text: "test button",
            size: "small",
            color: "red",
            variation: "link",
            onClick: jest.fn()
        }
        wrapper = mount(<Button {...props} />);
    })
    it('should render component', () => {
        expect(renderer.create(<Button {...props} />).toJSON()).toMatchSnapshot();
    });
    it('check class name props', () => {
        expect(wrapper.render().hasClass('button-link')).toBeTruthy();
        expect(wrapper.render().hasClass('button-link--red')).toBeTruthy();
        expect(wrapper.render().hasClass('button-link--small')).toBeTruthy();
        expect(wrapper.render().hasClass('button--small')).toBeFalsy();
    });
    it('on click function should not be called', () => {
        expect(props.onClick.mock.calls.length).toBe(0);
    });
    it('on click function should be called', () => {
        wrapper.find('button').simulate('click');
        expect(props.onClick).toHaveBeenCalled();
    });
});