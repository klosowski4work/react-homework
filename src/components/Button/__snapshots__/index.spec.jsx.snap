// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render component 1`] = `
<button
  className="button-link button-link--small  button-link--red"
  onClick={[Function]}
  type="button"
>
  test button
</button>
`;
